This does not just fix on macos, but also fixes on i686 and arm-32bit
but somehow only on glibc, not musl. Why?

From c7f5dcd953ff1fdfa002a8bccfb43eafcc6fddd4 Mon Sep 17 00:00:00 2001
From: trinity-1686a <trinity@deuxfleurs.fr>
Date: Sun, 15 Oct 2023 17:57:27 +0200
Subject: [PATCH] fix compilation on macos

fsblkcnt_t is ony 32b there, so we have to do an additional cast
---
 src/rpc/system.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rpc/system.rs b/src/rpc/system.rs
index 78fcc74b..4b40bec4 100644
--- a/src/rpc/system.rs
+++ b/src/rpc/system.rs
@@ -899,8 +899,8 @@ impl NodeStatus {
 		use nix::sys::statvfs::statvfs;
 		let mount_avail = |path: &Path| match statvfs(path) {
 			Ok(x) => {
-				let avail = x.blocks_available() * x.fragment_size() as u64;
-				let total = x.blocks() * x.fragment_size() as u64;
+				let avail = x.blocks_available() as u64 * x.fragment_size() as u64;
+				let total = x.blocks() as u64 * x.fragment_size() as u64;
 				Some((x.filesystem_id(), avail, total))
 			}
 			Err(_) => None,
-- 
2.40.1

